indexing-service/src/main/java/org/apache/druid/indexing/common/task/CompactionTask.java

```diff
@@ -671,7 +671,10 @@ private static DataSchema createDataSchema(
         finalDimensionsSpec,
         finalMetricsSpec,
         uniformGranularitySpec,
-        transformSpec == null ? null : new TransformSpec(transformSpec.getFilter(), null)
+        transformSpec == null ? null : new TransformSpec(transformSpec.getFilter(), null),
+        existingSegmentAnalyzer.getHasRolledUpSegments(),
+        null,
+    null
```

Comment:
Should we pass a non-null `ObjectMapper` here?
The logic in `DataSchema` is a little involved and I am not sure if the `ObjectMapper` will ever be used but better to be on the safe side than have to deal with an NPE that we can't decipher.

Reply:
It was passed as `null` even earlier, so this is just refactoring. But if you think we should still pass non-null -- which does make sense --  then I can update it.

Reply:
No, if it was already `null`, then let's continue passing `null`.